pkg/collector/skipper_collector.go

```diff
@@ -128,26 +134,65 @@ func getWeights(ingressAnnotations map[string]string, backendAnnotations []strin
 	return 0.0, errBackendNameMissing
 }
 
-// getCollector returns a collector for getting the metrics.
-func (c *SkipperCollector) getCollector() (Collector, error) {
-	ingress, err := c.client.NetworkingV1beta1().Ingresses(c.objectReference.Namespace).Get(context.TODO(), c.objectReference.Name, metav1.GetOptions{})
-	if err != nil {
-		return nil, err
+func getRouteGroupWeight(backends []rgv1.RouteGroupBackendReference, backendName string) (float64, error) {
+	if len(backends) <= 1 {
+		return 1.0, nil
 	}
 
-	backendWeight, err := getWeights(ingress.Annotations, c.backendAnnotations, c.backend)
-	if err != nil {
-		return nil, err
+	for _, backend := range backends {
+		if backend.BackendName == backendName {
+			return float64(backend.Weight) / 100.0, nil
+		}
 	}
-	config := c.config
 
+	if backendName != "" {
+		return 0.0, nil
+	}
+
+	return 0.0, errBackendNameMissing
+}
+
+// getCollector returns a collector for getting the metrics.
+func (c *SkipperCollector) getCollector(ctx context.Context) (Collector, error) {
 	var escapedHostnames []string
-	for _, rule := range ingress.Spec.Rules {
-		escapedHostnames = append(escapedHostnames, regexp.QuoteMeta(strings.Replace(rule.Host, ".", "_", -1)))
+	var backendWeight float64
+	switch c.objectReference.Kind {
+	case "Ingress":
+		ingress, err := c.client.NetworkingV1beta1().Ingresses(c.objectReference.Namespace).Get(ctx, c.objectReference.Name, metav1.GetOptions{})
+		if err != nil {
+			return nil, err
+		}
+
+		backendWeight, err = getWeights(ingress.Annotations, c.backendAnnotations, c.backend)
```

Comment:
:nail_care: nit: rename to 
```suggestion
		backendWeight, err = getIngressWeight(ingress.Annotations, c.backendAnnotations, c.backend)
```

Reply:
changed!